test/test_psutil.py

```diff
@@ -3231,6 +3252,116 @@ def test_proc_open_files(self):
+                                   stdin=subprocess.PIPE,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT)
+        p = psutil.Process(subp.pid)
+        self.assertIsInstance(p.name(), str)
+        self.assertEqual(encode_path(os.path.basename(p.name())), b"\xc0\x80")
+        subp.communicate()
+        self.assertEqual(subp.returncode, 0)
+
+    def test_proc_cmdline(self):
+        funny_file = os.path.join(self.temp_directory, b"\xc0\x80")
+        open(funny_file, "wb").close()
+        self.addCleanup(safe_remove, funny_file)
+        cmd = [self.test_executable]
+        if WINDOWS:
+            cmd.extend(["/K", "type \xc0\x80"])
+        else:
+            cmd.extend([b"\xc0\x80", b"-"])
+        subp = get_test_subprocess(cmd=cmd,
+                                   stdin=subprocess.PIPE,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT,
+                                   cwd=decode_path(self.temp_directory))
+        p = psutil.Process(subp.pid)
+        self.assertEqual(p.cmdline()[1:], cmd[1:])
+        subp.communicate()
+        self.assertEqual(subp.returncode, 0)
+
+    def test_proc_cwd(self):
+        funny_directory = os.path.join(self.temp_directory, b"\xc0\x80")
+        os.mkdir(funny_directory)
+        self.addCleanup(safe_rmdir, funny_directory)
+        subp = get_test_subprocess(cmd=[self.test_executable],
+                                   stdin=subprocess.PIPE,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT,
+                                   cwd=decode_path(funny_directory))
+        p = psutil.Process(subp.pid)
+        self.assertEqual(encode_path(p.cwd()), funny_directory)
+        subp.communicate()
+        self.assertEqual(subp.returncode, 0)
+
+    @unittest.skipIf(WINDOWS, "does not work on windows")
+    def test_proc_open_files(self):
+        funny_file = os.path.join(self.temp_directory, b"\xc0\x80")
+        test_script = os.path.join(self.temp_directory, b"test.py")
+        with open(test_script, "wt") as f:
+            f.write(textwrap.dedent(r"""
+            import sys, os
+            with open(%r, "wb") as f1, open(__file__, "rb") as f2:
+                sys.stdin.read()
+            """ % funny_file))
+        self.addCleanup(safe_remove, test_script)
+        subp = get_test_subprocess(cmd=[PYTHON, decode_path(test_script)],
+                                   stdin=subprocess.PIPE,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT)
+        self.addCleanup(safe_remove, funny_file)
+        p = psutil.Process(subp.pid)
+        # wait for the file to appear
+        while len(os.listdir(self.temp_directory)) == 1:
+            time.sleep(0.01)
```

Comment:
You can use `call_until(os.listdir(self.temp_directory), "len(ret) == 1")` instead. The advantage is that it will timeout after a while in case the directory doesn't change.